sway-lsp/src/utils/markdown.rs

```diff
@@ -0,0 +1,179 @@
+//! Transforms markdown
+const SWAYDOC_FENCES: [&str; 2] = ["```", "~~~"];
+
+pub(crate) fn format_docs(src: &str) -> String {
+    let mut processed_lines = Vec::new();
+    let mut in_code_block = false;
+    let mut is_sway = false;
+
+    for mut line in src.lines() {
+        if in_code_block && is_sway && code_line_ignored_by_swaydoc(line) {
+            continue;
+        }
+
+        if let Some(header) = SWAYDOC_FENCES
+            .into_iter()
+            .find_map(|fence| line.strip_prefix(fence))
+        {
+            in_code_block ^= true;
+
+            if in_code_block {
+                is_sway = is_sway_fence(header);
+
+                if is_sway {
+                    line = "```sway";
+                }
+            }
```

Comment:
should `is_sway` be reset to false if `in_code_block` is false?

Reply:
This function accounts for different types of code blocks that may or may not be `sway` code. I also borrowed this function from rust-analyzer. It seems to be working as intended. Good spot though.